### PR TITLE
Missing Unit Tests

### PR DESCRIPTION
**Pull request by viraj124**
- unit tests for admin and user withdraw
- event update in polygon contract for graph
- valid-user check added in contrat

### CODE REVIEW

File: test/GoodGhosting.test.js [hunk 1]
```diff
@@ -1200,6 +1359,44 @@ contract("GoodGhosting", (accounts) => {
             await goodGhosting.redeemFromExternalPool({ from: player1 });
             await truffleAssert.reverts(goodGhosting.adminFeeWithdraw({ from: admin }), "No Fees Earned");
         });
+
+        it("withdraw fees when there's only interest generated by external pool", async () => {
```
Comment (fodisi) [inline test/GoodGhosting.test.js:1363]:
```suggestion
        it("withdraw fees if there are no winners, but there's interest generated by external pool", async () => {
```

**Review by fodisi — CHANGES_REQUESTED** [review]
looking great. We just need to also cover these 2 scenarios for the section when adminFee is set to zero:

- early withdrawal fees only (no interest from external pool)
- early withdrawal fees and interest from external pool

File: test/GoodGhosting.test.js (hunk 1 above)
Comment (viraj124) [reply test/GoodGhosting.test.js:1363]: @fodisi covered in this commit https://github.com/Good-Ghosting/goodghosting-smart-contracts/pull/114/commits/9cd39affa29ed5af43f8b4a7e53cb749a4a41d87

Comment (fodisi) [reply test/GoodGhosting.test.js:1363]: @viraj124 
the commit you referred to covers from a player perspective. my comment is related to the admin perspective when there are no winners. Thanks

Comment (viraj124) [reply test/GoodGhosting.test.js:1363]: @fodisi 
also there https://github.com/Good-Ghosting/goodghosting-smart-contracts/blob/refactor/missing-unit-tests/test/GoodGhosting.test.js#L1167

**Review by fodisi — APPROVED** [review]

**Pull request merged**